Sources/Widgets/Widgets3D/LineWidget/index.js

```diff
@@ -132,39 +145,20 @@ function vtkLineWidget(publicAPI, model) {
   // --- Public methods -------------------------------------------------------
 
   publicAPI.getDistance = () => {
-    const nbHandles =
-      model.widgetState.getHandle1List().length +
-      model.widgetState.getHandle2List().length;
-    if (nbHandles !== 2) {
+    const nbHandles = model.widgetState.getNbHandles();
+    if (nbHandles < 2) {
       return 0;
     }
     return Math.sqrt(
       distance2BetweenPoints(
-        model.widgetState.getHandle1List()[0].getOrigin(),
-        model.widgetState.getHandle2List()[0].getOrigin()
+        model.widgetState.getHandle1().getOrigin(),
+        model.widgetState.getHandle2().getOrigin()
       )
     );
   };
 
-  publicAPI.updateTextValue = (text) => {
-    if (typeof model.widgetState.getText() !== 'undefined')
-      model.widgetState.getText().setText(text);
-  };
-
-  publicAPI.updateTextProps = (input, prop) => {
-    if (prop === 'positionOnLine') {
-      publicAPI.setPositionOnLine(input / 100);
-    }
-    updateTextPosition(model, publicAPI.getPositionOnLine());
-    model.widgetState.setPositionOnLine(publicAPI.getPositionOnLine());
-  };
-
-  publicAPI.updateHandleFromUI = (input, handleId) => {
-    if (handleId === 1) {
-      model.handle1Shape = input;
-    } else if (handleId === 2) {
-      model.handle2Shape = input;
-    }
+  publicAPI.setHandleShape = (handleId, shape) => {
+    model[`handle${handleId}Shape`] = shape;
```

Comment:
Use an array instead:
model.handle${handleId}Shape -> model.shapes[handleId] = shape

Reply:
Why do i need an array here? Is it for readability or performances purposes? Shouldn't I use a setter function instead?

Reply:
for readability. Performance is rarely a good reason in JS.

Btw, I think this should be stored in the widget state instead of the model.